### PR TITLE
Introduce setters for opset7::Gelu

### PR DESCRIPTION
**Pull request by jane-intel**
### Tickets:
 - *90601*

### CODE REVIEW

File: src/core/include/openvino/op/gelu.hpp [hunk 1]
```diff
@@ -13,9 +13,9 @@ namespace v0 {
 /// \brief Gaussian Error Linear Unit
 /// f(x) = 0.5 * x * (1 + erf( x / sqrt(2) )
 /// \ingroup ov_ops_cpp_api
-class OPENVINO_API Gelu : public Op {
+class OPENVINO_API Gelu : public util::UnaryElementwiseArithmetic {
```
Comment (rkazants) [inline src/core/include/openvino/op/gelu.hpp:16]: It is a sort of API change. Looks not safe since it becomes ABI incompatible

Comment (jane-intel) [reply src/core/include/openvino/op/gelu.hpp:16]: OV doesn't claim ABI compatibility.
Discussed with @ilya-lavrenov and @ilyachur

Comment (ilyachur) [reply src/core/include/openvino/op/gelu.hpp:16]: OpenVINO doesn't have requirements for ABI compatibility. It is a valid change. And no problem here.

File: src/core/include/openvino/op/gelu.hpp [hunk 2]
```diff
@@ -66,6 +66,9 @@ class OPENVINO_API Gelu : public util::UnaryElementwiseArithmetic {
     std::shared_ptr<Node> clone_with_new_inputs(const OutputVector& new_args) const override;
 
     GeluApproximationMode get_approximation_mode() const;
+    void set_approximation_mode(const GeluApproximationMode& approximation_mode) {
```
Comment (ilyachur) [inline src/core/include/openvino/op/gelu.hpp:69]: Can we add tests for all such changes in order to cover getters and setters?

File: src/core/include/openvino/op/gelu.hpp (hunk 1 above)
Comment (rkazants) [reply src/core/include/openvino/op/gelu.hpp:16]: This change is logical, but changing header files in public API is not preferable.

**Review by sshlyapn — APPROVED** [review]

**Pull request merged**